go/vt/sqlparser/utils_test.go

```diff
@@ -180,3 +181,85 @@ func TestQueryMatchesTemplates(t *testing.T) {
 		})
 	}
 }
+
+func TestReplaceTableQualifiers(t *testing.T) {
+	origDB := "_vt"
+	tests := []struct {
+		name    string
+		in      string
+		newdb   string
+		out     string
+		wantErr bool
+	}{
+		{
+			name:    "invalid select",
+			in:      "select frog bar person",
+			out:     "",
+			wantErr: true,
+		},
+		{
+			name: "simple select",
+			in:   "select * from _vt.foo",
+			out:  "select * from foo",
+		},
+		{
+			name:  "simple select with new db",
+			in:    "select * from _vt.foo",
+			newdb: "_vt_test",
+			out:   "select * from _vt_test.foo",
+		},
+		{
+			name:  "simple select with new db same",
+			in:    "select * from _vt.foo where id=1", // should be unchanged
+			newdb: "_vt",
+			out:   "select * from _vt.foo where id=1",
+		},
+		{
+			name:  "simple select with new db needing escaping",
+			in:    "select * from _vt.foo",
+			newdb: "1_vt-test",
+			out:   "select * from `1_vt-test`.foo",
+		},
+		{
+			name: "complex select",
+			in:   "select table_name, lastpk from _vt.copy_state where vrepl_id = 1 and id in (select max(id) from _vt.copy_state where vrepl_id = 1 group by vrepl_id, table_name)",
+			out:  "select table_name, lastpk from copy_state where vrepl_id = 1 and id in (select max(id) from copy_state where vrepl_id = 1 group by vrepl_id, table_name)",
+		},
+		{
+			name: "complex select",
+			in:   "select t1.col1, t2.col2 from _vt.t1 as t1 join _vt.t2 as t2 on t1.id = t2.id",
+			out:  "select t1.col1, t2.col2 from t1 as t1 join t2 as t2 on t1.id = t2.id",
+		},
+		{
+			name: "simple insert",
+			in:   "insert into _vt.foo(id) values (1)",
+			out:  "insert into foo(id) values (1)",
+		},
+		{
+			name: "simple update",
+			in:   "update _vt.foo set id = 1",
+			out:  "update foo set id = 1",
+		},
+		{
+			name: "simple delete",
+			in:   "delete from _vt.foo where id = 1",
+			out:  "delete from foo where id = 1",
+		},
+		{
+			name: "simple set",
+			in:   "set names 'binary'",
+			out:  "set names 'binary'",
+		},
```

Comment:
Can you please add tests with:
- a derived table 
- an inner select from another table

Reply:
Done!